### ISSUE TITLE
Soft blending is not differentiable

### ISSUE THREAD

**Issue by Jiayi-Wang-MPI** [facebookresearch/pytorch3d#61]
The current implementations of sigmoid_alpha_blend and softmax_rgb_blend in blending.py are not differentiable. This is due the to the replacement of cumulative product with the exponential of log sums for the alpha calculation. When prob_map is 1.0, the gradient is not defined. 

```
# alpha = 1.0 - torch.cumprod((1.0 - prob), dim=-1)[..., -1]
alpha = 1.0 - torch.exp(torch.log((1.0 - prob_map)).sum(dim=-1))
```

**Comment by nikhilaravi** [comment]
@Jiayi-Wang-MPI thanks for raising this issue. You can replace this line with the line above - we noticed that `torch.cumprod` backward was not very stable as explained in the comment in the code. Creating C++/CUDA implementations of these blending functions to improve stability is on our roadmap.

**Comment by nikhilaravi** [comment]
@Jiayi-Wang-MPI  you could also try this: `alpha = 1.0 - torch.prod((1.0 - prob_map), dim=-1)`.  We will be updating the blending functions soon.

**Comment by Jiayi-Wang-MPI** [comment]
@nikhilaravi thanks for the help! The suggested solutions seem to work for now. I am looking forward to the more stable C++/CUDA implementation!

**Comment by nikhilaravi** [comment]
@Jiayi-Wang-MPI the fix I mentioned above has been merged into master and the backward pass of soft blending  should now be stable. I am closing this issue. If you continue to experience any problems feel free to reopen it.
